### PR TITLE
Cleanup the View code logic and add comments.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

File: torch_xla/csrc/tensor.cpp [hunk 1]
```diff
@@ -198,7 +198,7 @@ xla::int64 XLATensor::GetUniqueId() const { return data()->unique_id; }
 std::shared_ptr<xla::ComputationClient::Data> XLATensor::GetXlaData() {
   bool up_to_date = true;
   if (data()->view != nullptr) {
-    auto ir_node_updated = GetViewIrNode(data()->view.get());
+    ViewIrNode ir_node_updated = GetViewIrNode(data()->view.get());
```
Comment (asuhan) [inline torch_xla/csrc/tensor.cpp]: Is the explicit type really needed here? It's pretty obvious what a `GetViewIrNode` is supposed to return.

Comment (dlibenzi) [reply torch_xla/csrc/tensor.cpp]: I like to reserve auto for things which are templated.
Having a type explicit avoid looking up the API to figure out.

**Comment by dlibenzi** [comment]
I think the CircleCI failed with an abort() in the C++ tests, like you noticed sometimes Alex.

**Comment by asuhan** [comment]
Yes, it's a flaky on-exit crash. Let's ignore it for now.

**Comment by dlibenzi** [comment]
Is there a way to re-trigger CircleCI workflows manually?

**Pull request merged**